### PR TITLE
✨🔥🐛🐛 Code Polish

### PR DESCRIPTION
**Pull request by anarmawala**
⬆ Upgrade dependencies / `rush` and `pnpm` versions
🐛 Ask notification permission only when needed
🐛 Set `secure` cookie to `auto`
🔻 Reduce package sizes / build times
🐛 Fix `null` bug on `api`

### CODE REVIEW

**Comment by bmiddha** [comment]
🏗️ Update Client build config
🐳 Update Docker configs
👷 Update GitHub Action Workflows
➖ Remove deprecated dependencies
⬆️ Bump Dependencies
✨ Add new logo

**Review by bmiddha — APPROVED** [review]
LGTM 👍

**Comment by bmiddha** [comment]
🐳 Moved to Azure Container Registry
✏️ Rebranding initial pass

File: api/src/util/notifier.ts [hunk 1]
```diff
@@ -26,28 +26,22 @@ export const notifyUser = async (user: UserObject, classData: any) => {
...
-            const msg = {
-                to: user.email,
-                from: process.env.SENDGRID_EMAIL_ADDRESS,
-                subject: `TRACKING CLASS CRN ${courseReferenceNumber} HAS STATUS: ${statusMessage}`,
-                text: 'Thanks for using Registrum!',
-                html: '<strong>Thanks for using Registrum!</strong>'
-            }
-
-            // // * SEND EMAIL
+            // const msg = {
```
Comment (alexchomiak) [inline api/src/util/notifier.ts:30]: Why comment out the email logic? Are we migrating to a postfix solution?

**Review by alexchomiak — APPROVED** [review]
Everything else looks good! 💯 🌮

**Pull request merged**